JustLog/Classes/Log.swift

```diff
@@ -0,0 +1,129 @@
+//  Log.swift
+
+public enum LogType: String {
+    case debug
+    case warning
+    case verbose
+    case error
+    case info
+}
+
+public struct LogKeys {
+    
+    public static var logTypeKey: String = "log_type"
+    public static var appBundleIDKey = "app_bundle_ID"
+    public static var appVersionKey = "app_version"
+    public static var deviceTimestampKey = "device_timestamp"
+    public static var deviceTypeKey = "ios_device"
+    public static var errorCodeKey = "error_code"
+    public static var errorDomainKey = "error_domain"
+    public static var errorsKey = "errors"
+    public static var fileKey: String = "file"
+    public static var functionKey = "function"
+    public static var iosVersionKey = "ios_version"
+    public static var lineKey = "line"
+    public static var messageKey = "message"
+    public static var metadataKey = "metadata"
+    public static var userInfoKey = "user_info"
+}
+
+public protocol Loggable {
+    
+    static var defaultUserInfo: [String : Any]? { get set }
+    static var keys: LogKeys.Type { get set }
+    
+    var type: LogType { get }
+    var message: String { get }
+    var error: NSError? { get }
+    var customData: [String: Any]? { get set }
+    var file: String { get }
+    var function: String { get }
+    var line: UInt { get }
+    
+    func formattedMessage() -> String
+}
+
+public struct Log: Loggable {
+    
+    public static var defaultUserInfo: [String : Any]?
```

Comment:
More so here regarding static vars. Best in experiment with inheritance from a design perspective.